.github/workflows/build-test.yml

```diff
@@ -0,0 +1,235 @@
+
+      # install build dependencies
+      - name: Install ${{ inputs.compiler }} if needed
+        if: startsWith(inputs.compiler, 'clang-') || startsWith(inputs.compiler, 'gcc-')
+        run: |
+          sudo apt-get -q update
+          sudo apt-get -y install ${{ inputs.compiler }}
+
+      # - name: Install libprotobuf-c
+      #   run: |
+      #     sudo apt-get -q update
+      #     sudo apt-get -y install libprotobuf-c-dev
+
+      # if we have already built, unarchive and no need to configure/build with
+      # cmake
+      - name: Download previous build artifact
+        id: download-prior-build
+        continue-on-error: true
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ steps.configure-artifact.outputs.name }}
+          path: .
+
+      - name: Unarchive build artifact
+        if: steps.download-prior-build.outcome == 'success'
+        run: tar -xzf build.tar.gz
+
+      # otherwise, configure cmake, build, archive and upload
+      - name: CMake
+        if: steps.download-prior-build.outcome != 'success'
+        run: |
+          mkdir -p build
+          cd build
+          cmake .. ${{ steps.cmake-flags.outputs.flags }}
+          make rebuild_cache && make
+
+      - name: "Rebuild the list of tests to match the compile flags"
+        if: steps.download-prior-build.outcome != 'success'
+        working-directory: ./build
+        run: |
+          ./test/testsuite
+          if [[ $(diff list.txt ../test/list.txt; echo $?) != 0 ]]; then
+            mv list.txt ../test/list.txt
+            make rebuild_cache
+          fi
+
+      - name: Archive build artifact
+        if: steps.download-prior-build.outcome != 'success'
+        run: |
+          tar -czf build.tar.gz build
+
+      - name: Upload build artifact
+        if: steps.download-prior-build.outcome != 'success'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.configure-artifact.outputs.name }}
+          path: build.tar.gz
+
+      # testing
+
+      - name: "Download nats-server version ${{ inputs.server_version }}"
+        if: inputs.quick_test != 'ON' && !inputs.server_version
+        working-directory: ./build
+        run: |
+          if [[ "${{ inputs.server_version }}" == "latest" ]]; then
+            rel=$(curl -s https://api.github.com/repos/nats-io/nats-server/releases/latest | jq -r '.tag_name')
+          else
+            rel=${{ inputs.server_version }}
+          fi
+          wget https://github.com/nats-io/nats-server/releases/download/${rel}/nats-server-${rel}-linux-amd64.tar.gz
+          if [[ $? == 0 ]]; then
+            tar -xzf nats-server-${rel}-linux-amd64.tar.gz
+            cp nats-server-${rel}-linux-amd64/nats-server ../deps/nats-server
+          else
+            curl -sf 'https://binaries.nats.dev/nats-io/nats-server/v2?os=linux&arch=amd64&version=${{ inputs.server_version }}'' | PREFIX=. sh
```

Comment:
Thanks for giving this a try :) This url autodetects the os version and arch so can be used like this, for example:

```
curl -sf https://binaries.nats.dev/nats-io/nats-server/v2@dev | PREFIX=. sh
curl -sf https://binaries.nats.dev/nats-io/nats-server/v2@main | PREFIX=. sh
curl -sf https://binaries.nats.dev/nats-io/nats-server/v2@v2.9.19 | PREFIX=. sh
```